src/SFA.DAS.AANHub.Application/Apprentices/Commands/PatchApprenticeMember/PatchApprenticeMemberCommandValidator.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.ObjectModel;
+using FluentValidation;
+using SFA.DAS.AANHub.Application.Common.Validators.RequestedByMemberId;
+using SFA.DAS.AANHub.Domain.Interfaces.Repositories;
+
+namespace SFA.DAS.AANHub.Application.Apprentices.Commands.PatchApprenticeMember
+{
+    public class PatchApprenticeMemberCommandValidator : AbstractValidator<PatchApprenticeMemberCommand>
+    {
+        public const string Replace = "replace";
+        public const string NoPatchOperationsPresentErrorMessage = "There are no patch operations in this call";
+        public const string PatchOperationContainsUnavailableFieldErrorMessage = "This patch operation contains an unexpected field and will not continue";
+        public const string PatchOperationContainsUnavailableOperationErrorMessage = "This patch operation contains an unexpected operation and will not continue";
+
+        public static readonly IList<string> PatchFields = new ReadOnlyCollection<string>(
+                new List<string>
+                {
+                    "Email",
+                    "Name"
+                });
+
+        public PatchApprenticeMemberCommandValidator(IMembersReadRepository membersReadRepository, IApprenticesReadRepository apprenticesReadRepository)
+        {
+            Include(new RequestedByMemberIdValidator(membersReadRepository));
+
+            RuleFor(c => c.Patchdoc.Operations.Count).GreaterThan(0).WithMessage(NoPatchOperationsPresentErrorMessage);
+
+            RuleFor(c => c.Patchdoc.Operations.Count(operation => !operation.op.Equals(Replace, StringComparison.CurrentCultureIgnoreCase)))
+                .Equal(0)
+                .WithMessage(PatchOperationContainsUnavailableOperationErrorMessage);
+
+            RuleFor(c => c.Patchdoc.Operations.Count(operation => !PatchFields.Contains(operation.path)))
+                .Equal(0)
+                .WithMessage(PatchOperationContainsUnavailableFieldErrorMessage);
+
+            RuleFor(x => x.ApprenticeId)
+                .NotNull()
+                .NotEmpty();
+
+            RuleFor(x => x.Email)
```

Comment:
I would say that Email and Name probably need to be in some kind of "Base patch validator" like we have for the Create route? I imagine all subsequent calls will need to validate this data too?

Reply:
Not sure of this because of the `when` pattern below.

```
            RuleFor(x => x.Email)
                .NotEmpty()
                _.When(c => string.IsNullOrEmpty(c.Name))_
                .MaximumLength(256)
                .Matches(Constants.RegularExpressions.EmailRegex);

            RuleFor(c => c.Name)
                .NotEmpty()
                _.When(c => string.IsNullOrEmpty(c.Email))_
                .MaximumLength(200);
```
also Employers make use of Name, Email and Organisation. Partner has Email and Organisation.

Reply:
@DevenKShah, @LiamWright, if we are happy with my above reply we are good for a final review as the code has changed once again.

Reply:
@ravihejmadi why do we have a `When` condition? These fields should be validated irrespectively.

Reply:
Because Name and Email are non mutually exclusive. 
The requirement is that if Name is null/empty then Email must not be empty. Similarly if Email is null/empty then Name must not be empty